packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -434,6 +470,7 @@ export type SdkModelPropertyType =
   | SdkCookieParameter;
 
 export interface MultipartOptions {
+  name: string;
```

Comment:
I remember from @msyyc that the `name` (field name) of a part could be undefined, in some cases (multipart/mixed ?).

Are we able to know in which case, the `name` is provided by TCGC, instead of a real serialized name on wire?

Reply:
the name is either from the property name or from the http part name. let me double check with @msyyc.

Reply:
Each field has a name. As @tadelesh said, it is either from property name or http part name. 
![image](https://github.com/user-attachments/assets/66a34e09-c256-4a5a-b6e6-997925e8de02)

Reply:
Is it doc for multipart/form-data or multipart/mixed?

![image](https://github.com/user-attachments/assets/b6cc0fea-0ab9-4131-a427-8e7e33fe9c80)

I am fine TCGC only support form-data. In such case, I expect it output diagnostic error on mixed.

Reply:
I see. I make an issue to track it https://github.com/Azure/typespec-azure/issues/2065